x-pack/plugins/apm/typings/anomaly_detection.ts

```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface AnomalyDetectionJobByEnv {
+  'service.environment': string;
```

Comment:
As mentioned elsewhere: this might be simpler as `environment`
```suggestion
  environment: string;
```